core/src/main/kotlin/org/sert2521/sertain/motors/TalonMotorController.kt

```diff
@@ -0,0 +1,185 @@
+package org.sert2521.sertain.motors
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX
+
+class TalonMotorController(
+        id: TalonId,
+        vararg followerIds: MotorId,
+        configure: MotorController.() -> Unit = {}
+) : MotorController() {
+    val srx = TalonSRX(id.number)
+
+    var master: MotorController? = null
+        set(value) {
+            if (value != null) {
+                srx.follow(value.baseController!!)
+            }
+            field = value
+        }
+
+    val followers: MutableMap<MotorId, MotorController> = with(mutableMapOf<MotorId, MotorController>()) {
+        followerIds.forEach {
+            set(it, motorController(it)!!.apply {
+                master = this
+            })
+        }
+        toMutableMap()
+    }
+
+    override val baseController = srx
+
+    override fun eachMotor(configure: MotorController.() -> Unit) {
+        apply(configure)
+        eachFollower(configure)
+    }
+
+    override fun eachFollower(configure: MotorController.() -> Unit) {
+        followers.forEach {
+            it.value.apply(configure)
+        }
+    }
+
+    override val controlMode: ControlMode
+        get() = when (srx.controlMode) {
+            com.ctre.phoenix.motorcontrol.ControlMode.PercentOutput -> ControlMode.PERCENT_OUTPUT
```

Comment:
Been a while since I've done this in Kotlin, but I think you can:

```kotlin
import com.ctre.phoenix.motorcontrol.ControlMode as PhoenixControlMode
```

and then do this here 

```suggestion
            PhoenixControlMode.PercentOutput -> ControlMode.PERCENT_OUTPUT
```